src/coreclr/vm/ceeload.cpp

```diff
@@ -220,6 +220,14 @@ void Module::UpdateNewlyAddedTypes()
     DWORD countExportedTypesAfterProfilerUpdate = GetMDImport()->GetCountWithTokenKind(mdtExportedType);
     DWORD countCustomAttributeCount = GetMDImport()->GetCountWithTokenKind(mdtCustomAttribute);
 
+    if (m_dwTypeCount == countTypesAfterProfilerUpdate
```

Comment:
Is it possible to do something even cheaper as a first check to see if there is even a profiler attached?  The profiler initialization has already triggered when the first call to this function is made, so I'm wondering if we can do an "is profiler attached" check before any of these function calls?

Reply:
I don't think we can skip all that much of it. In Module::NotifyProfilerLoadFinished we have to cache m_dwTypeCount, etc so if a profiler attaches later in the process we know how many types we started out with. The profiler can call Module::ApplyMetadata after making changes at any point.

We could check to see if a profiler was called at all in NotifyProfilerLoadFinished and skip the call to UpdateNewlyAddedTypes if the profiler wasn't called.

Reply:
That sounds great!